src/templates/components/datagrid.html

```diff
@@ -30,9 +39,11 @@
       </td>
     </tr>
   </table>
-  <div class="datagrid-add" ng-if="!component.hasOwnProperty('validate') || !component.validate.hasOwnProperty('maxLength') || rows.length < component.validate.maxLength">
-    <a ng-click="(readOnly || formioForm.submitting)? null: addRow() " ng-disabled = "readOnly || formioForm.submitting" class="btn btn-primary">
+  <div style="margin-top: 10px" class="datagrid-add" ng-if="!component.hasOwnProperty('validate') || !component.validate.hasOwnProperty('maxLength') || rows.length < component.validate.maxLength">
+    <a ng-if="component.addAnotherPosition === 'bottom' || component.addAnotherPosition === 'both'" ng-click="(readOnly || formioForm.submitting)? null: addRow() " ng-disabled = "readOnly || formioForm.submitting" class="btn btn-primary">
```

Comment:
What about legacy support? Won't this make the original button disappear if they don't have component.addAnotherPosition set on it?

Reply:
Good Catch, Addressed and moved the logic to the parent div to prevent if from loading when not needed.